identity/src/lib/rpc/server.rs

```diff
@@ -41,11 +41,7 @@ impl IdentityServer {
 impl IdentityService for IdentityServer {
     /// Returns an user
     async fn get_user(self, _: context::Context, user_id: u32) -> RpcResult<User> {
-        use crate::db::schema::users::dsl;
-
-        let result = dsl::users
-            .find(user_id as i32)
-            .first::<models::User>(&self.get_db());
+        let result = queries::get_user(user_id as i32, &self.get_db());
```

Comment:
I think you can simplify this function (and the other endpoints) even further, by using the `?` operator:
```rust
    async fn get_user(self, _: context::Context, user_id: u32) -> RpcResult<User> {
        let user = queries::get_user(user_id as i32, &self.get_db())?;

        Ok(User {
            id: user.id,
            sub: user.sub,
            email: user.email,
            given_name: user.given_name,
            family_name: user.family_name,
            picture: user.picture,
            active: user.active,
        })
    }
```

If you then also implement `From<db_models::User> for rpc_models::User` then you can even have it like:
```rust
    async fn get_user(self, _: context::Context, user_id: u32) -> RpcResult<User> {
        let user = queries::get_user(user_id as i32, &self.get_db())?;
        Ok(user.into())
    }
```

Reply:
Yes, I thought about this too. Then decided to first have testing, then simplify and then optimize.

Reply:
Sure, but you are already touching the queries anyways so I guess adding the `?` is not too much effort.